code/_onclick/ai_onclick.dm

```diff
@@ -212,8 +212,11 @@
 /obj/machinery/door/airlock/AICtrlClick(mob/living/silicon/user) // Bolts doors
 	if(!ai_control_check(user))
 		return
-	if(user.can_instant_lockdown() || do_after_once(user, 3 SECONDS, needhand = FALSE, target = src, allow_moving = TRUE, attempt_cancel_message = "Bolting [src] cancelled.", special_identifier = "Bolt"))
-		toggle_bolt(user)
+	if(locked)
+		toggle_bolt(user) // Un-bolt immediately.
+	else
+		if(user.can_instant_lockdown() || do_after_once(user, 3 SECONDS, needhand = FALSE, target = src, allow_moving = TRUE, attempt_cancel_message = "Bolting [src] cancelled.", special_identifier = "Bolt"))
+			toggle_bolt(user) // Bolt after a delay unless emagged.
```

Comment:
```suggestion
	if(locked)
		toggle_bolt(user) // Un-bolt immediately.
		return
	if(user.can_instant_lockdown() || do_after_once(user, 3 SECONDS, needhand = FALSE, target = src, allow_moving = TRUE, attempt_cancel_message = "Bolting [src] cancelled.", special_identifier = "Bolt"))
		toggle_bolt(user) // Bolt after a delay unless emagged.
```

Reply:
Actually, I'd suggest just moving `locked` into the or condition, similar to `user.can_instant_lockdown()`.